### ISSUE TITLE
net::ERR_CONNECTION_REFUSED with OAuth 

### ISSUE THREAD

**Issue by Heiner-Sch** [MarcusWolschon/osmeditor4android#1405]
I installed Vespucci on a new Android 11 phone.
If I try to configure OAuth via the toolbox I see the error message net::ERR_CONNECTION_REFUSED
Same error comes up if I try to send feedback.

But at the same time I can download OSM data and see the background from the Internet.

Found this line in the debug output: network enabled false
But there is the same line in my old Android 8 phone, which works without issues.

It's a fresh Vespucci installation. No additional presets installed

Any idea how to fix this?

## Vespucci Version
Vespucci 15.2.1

## Download source
f-droid

## Device (Manufacturer and Model)
Google Pixel 5

## Android Version 
Android 11 Security Update 05.05.2021

## Behaviour/Symptoms
error message net::ERR_CONNECTION_REFUSED

## Expected Behaviour
Browser (Firefox) opens with OSM logon screen.

## How to recreate
this happens always

## Crash dump submitted (no or yes + date)
no crash dump

Debug output:
Vespucci 15.2.1
Flavor: current
Maximum avaliable memory 536870912
Total memory used 104067136
Tile Cache MAPNIK usage Size 0 of maximum 67108864 #entries 0
Tile Cache ESRIWORLDIMAGERY usage Size 20447232 of maximum 67108864 #entries 78
State file size 33051 last changed 2021-06-03 08:02:10
Bug state file size 249 last changed 2021-06-03 08:02:10
Relations (current/API): 0/0
Ways (current/API): 44/0
Nodes (current/Waynodes/API): 579/577/0
Available location providers
passive enabled true
network enabled false
gps enabled true

## Any other potentially relevant information
Chrome is disabled
Firefox is default browser
SystemWebView is up to date

**Comment by simonpoole** [comment]
Hi, that both connecting to the OSM API and github doesn't work indicates that something is fundamentally wrong however there are essentially no commonalities between the two that would point to a specific issue. Android 11 as such is not known to cause issues at this point in time.

In particular you are connecting to the same servers for reading and writing from/to the OSM API so connectivity issues should affect both. Checking the API configuration (OSM data in the layer dialog "Configure") would still be a good thing to do to rule out install issues.

You can try to "Authorize OAuth" and then immediately send a manual crash dump if it goes wrong via the debug menu, but there is no guarantee that the log included with the dump will show anything relevant, but it is probably worth the try.

The "network enabled" line in the debug output refers to location data generated from WLAN and mobile cell tower locations, so doesn't have anything to do with the issue at hand.

**Comment by Heiner-Sch** [comment]
Now I'm Developer on my phone.
The WebView: Android System WebView 90.0.4430.210

I tried to use OAuth with Chrome. 
Chrome enabled, updated and standart browser. Same error.
Even an uninstall and install of Vespucci doesn't help.
OSM-API-URL is https://api.openstreetmap.org/api/0.6/ and it's checked.

The output of Debug doesn't show any news.

At the same time I can use OAuth with OSMand successful.

**Comment by simonpoole** [comment]
It is just the initial handshake that uses the webview, so osmand isn't really relevant they might to the handshake differently. 

Can you submit a crash dump immediately after trying to authenticate.

**Comment by Heiner-Sch** [comment]
Hope I've done it right.
Sent debug information with #1405

Or do you want an Android crash dump?

**Comment by simonpoole** [comment]
The information didn't help an awful lot, but the log does indicate that it is using org.chromium.android_webview.AwContents.r(chromium-TrichromeWebViewGoogle6432.aab-stable-443021034:2). 

Did you restart your device after you re-enabled chrome? The whole thing is a bit weird, as neither android 11 emulators running stock Android nor any real devices here with Android 11, show the behavior (including disabling Chrome).

**Comment by Heiner-Sch** [comment]
Chrome enabled, updated and standart browser, reboot. Same error.
Sent again debug information with #1405

About Google Chome:
Chrome 91.0.4472.88

I disabled some other Google apps, like on my Andoid 8 phone.
But Google Play, Play Store, Carrier Services and now also Chrome are active.

**Comment by simonpoole** [comment]
We should probably rule out that the issue is caused by F-Droid, could you install the most recent APK from here and test? This will require removing the F-Droid build.

**Comment by Heiner-Sch** [comment]
Vespucci from F-Droid, Play Store and the latest version here have all the same problem on my phone.

And I need to say, that I haven't authorised Oauth on my Android 8 phone in the last days.
I tested this now and got the same error there.
So there seems something wrong with the way I configure my Androids?!

A test with an older vesion from F-Droid didn't work also, but here I saw the OSM login screen for a second, before the authentication failed with the error message again.

I'm going to enable all Google apps step by step now, to see if this will fix the issue.
Any idea where to start?
Is it helpful to see the crash dump of the Android 8 phone also?
